### PR TITLE
Refactored metrics plugin and modified docs

### LINKED ISSUE #403: Metrics readme incorrectly claims that micrometer integrates with Spring Boot 1.x metrics
## Expected Behavior

Readme shouldn't claim that `/metrics` works for Spring Boot 1.x.

## Actual Behavior

Readme is incorrect.

### PR DESCRIPTION
**Pull request by whiskeysierra**
## Motivation and Context

Fixes #403

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.

### CODE REVIEW

File: riptide-metrics/src/main/java/org/zalando/riptide/metrics/MetricsPlugin.java [hunk 1]
```diff
@@ -10,33 +12,63 @@
...
-                        if (nonNull(response)) {
-                            measurement.record(response);
-                        }
-                    }));
+            final CompletableFuture<ClientHttpResponse> future;
+
+            try {
+                future = execution.execute();
+            } catch (final IOException e) {
+                measurement.record(null, e);
```
Comment (whiskeysierra) [inline riptide-metrics/src/main/java/org/zalando/riptide/metrics/MetricsPlugin.java]: How do I test this?

File: riptide-metrics/src/main/java/org/zalando/riptide/metrics/MetricsPlugin.java [hunk 2]
```diff
@@ -48,23 +80,54 @@ public RequestExecution prepare(final RequestArguments arguments, final RequestE
...
+
+    private String status(@Nullable final ClientHttpResponse response) {
+        if (response == null) {
+            return "CLIENT_ERROR";
         }
 
+        try {
+            return String.valueOf(response.getRawStatusCode());
+        } catch (final IOException e) {
+            return "IO_ERROR";
```
Comment (whiskeysierra) [inline riptide-metrics/src/main/java/org/zalando/riptide/metrics/MetricsPlugin.java]: Any idea how to cover/enforce this?

Comment (lukasniemeier-zalando) [reply riptide-metrics/src/main/java/org/zalando/riptide/metrics/MetricsPlugin.java]: Or we just let the exception propagate and don't measure in these cases

File: riptide-metrics/src/main/java/org/zalando/riptide/metrics/MetricsPlugin.java (hunk 1 above)
Comment (lukasniemeier-zalando) [reply riptide-metrics/src/main/java/org/zalando/riptide/metrics/MetricsPlugin.java]: super specific unit test with super specific mocks... :deciduous_tree:

**Comment by whiskeysierra** [comment]
I decided to get rid of the timing of failed requests, since I don't really see a big use case right now. We may need to revisit that decision.

**Comment by lukasniemeier-zalando** [comment]
:+1:

**Pull request merged**